plugin/trino-delta-lake/src/test/java/io/trino/plugin/deltalake/TestDeltaLakeConnectorTest.java

```diff
@@ -1753,4 +1753,253 @@ private Set<String> getAllFilesFromCdfDirectory(String tableName)
                 .filter(path -> path.contains("/" + CHANGE_DATA_FOLDER_NAME))
                 .collect(toImmutableSet());
     }
+
+    @Test
+    public void testPartitionFilterQueryNotDemanded()
+    {
+        String tableName = "test_partition_filter_not_demanded" + randomNameSuffix();
+
+        assertUpdate("CREATE TABLE %s (x VARCHAR, part VARCHAR) WITH (PARTITIONED_BY = ARRAY['part'])".formatted(tableName));
+
+        assertUpdate("INSERT INTO " + tableName + "(x, part) VALUES ('a', 'part_a'), ('b', 'part_b')", 2);
```

Comment:
Can we use this query public TestTable(SqlExecutor sqlExecutor, String namePrefix, String tableDefinition, List<String> rowsToInsert) - where we could specify the rows to insert.